core/connection/manager.go

```diff
@@ -166,32 +166,37 @@ func (manager *connectionManager) startConnection(consumerID identity.Identity,
 
 	messageChan := make(chan balance.Message, 1)
 
-	// TODO: load initial promise state
-	payments, err := manager.paymentIssuerFactory(promise.State{}, messageChan, dialog, consumerID, providerID)
+	consumerInfo := session.ConsumerInfo{
+		// TODO: once we're supporting payments from another identity make the changes accordingly
+		IssuerID:          consumerID,
+		MystClientVersion: metadata.VersionAsString(),
+	}
+
+	s, paymentInfo, err := session.RequestSessionCreate(dialog, proposal.ID, sessionCreateConfig, consumerInfo)
 	if err != nil {
 		return err
 	}
 
-	cancel = append(cancel, func() { payments.Stop() })
-
-	go manager.payForService(payments)
+	cancel = append(cancel, func() { session.RequestSessionDestroy(dialog, s.ID) })
 
-	consumerInfo := session.ConsumerInfo{
-		// TODO: once we're supporting payments from another identity make the changes accordingly
-		IssuerID:          consumerID,
-		MystClientVersion: metadata.VersionAsString(),
+	var promiseState promise.State
+	if paymentInfo != nil {
+		promiseState.Amount = paymentInfo.LastPromise.Amount
+		promiseState.Seq = paymentInfo.LastPromise.SequenceID
 	}
 
-	sessionID, sessionConfig, err := session.RequestSessionCreate(dialog, proposal.ID, sessionCreateConfig, consumerInfo)
+	payments, err := manager.paymentIssuerFactory(promiseState, messageChan, dialog, consumerID, providerID)
```

Comment:
`startConnection()` should start connection, I strongly disagree to grow it with unrelated logic.

Reply:
```
func (manager *connectionManager) Connect(
  ..
  manager.startConnection(consumerID, proposal, params)
  go manager.payForService(dialog, session, connectionSubscription)`
)

Reply:
refactored